### DISCUSSION TITLE
CP - Entries Relationship pulling too much data

### DISCUSSION THREAD

**Discussion by ryatkins** [statamic/cms#7128] [Troubleshooting]
On our Product page entry, we have multiple Entries relationships that pull from other collections to allow us to display data like the following:
- Related products (multiple)
- Disclaimer pages  (multiple)
- Product Category (single)

Using the Collection relationship it seems to pull all the data of each entry inside the collection.  The result is that it takes 20 seconds for each relationship to load, spikes the server CPU and takes forever to load entries.

How do I limit the data to just the ID and title of the entry?

Or do I use a different relationship type?

6 relationships, 5 out of 6 potentially pulling the same data as they are the same size
<img width="165" alt="image" src="https://user-images.githubusercontent.com/1211767/204157393-9a771203-5029-4cc8-ae15-22803938104b.png">

**Comment by ryatkins** [comment]
@jasonvarga - can you take a look at this and chime in on why my relationships take 20 seconds to load?

I just need a drop down of Entry Title and ID so that the relationship can be stored.

**Reply by jasonvarga** [reply]
Can you provide a repo for us to test this on? Thanks.

**Reply by ryatkins** [reply]
I've submitted a ticket through the support site with a Dropbox link to our site.

**Comment by mgtcampbell** [comment]
@ryatkins just wondering if you got anywhere with this issue?

I have a similar case where I have a website that loads a client's team member collection (they have 72 staff), but for each team member needs to load:

- Their job position where Positions is a taxonomy
- Their location(s) where Locations is a related collection via an Entries field type

Some other notes:

- I do have static caching set to half, which then loads the page quickly once it's all been processed once - however the site is frequently updated and therefore cache is often busted and Google seems to be picking up some of the _initial_ slow page load speeds before the cached version is set
- I could paginate the entries for faster load instead of loading all 72, however I have some on-page filters for location and position, so I would prefer to load them all at once so it's easy to implement dynamic javascript filtering without needing to get into complicated lazy loads etc

**Reply by aaronbushnell** [reply]
I wonder if this will be fixed in Statamic 5.x by PRs like #9317

**Reply by edalzell** [reply]
You can always try it!
